src/Adapter/MSTest.CoreAdapter/Discovery/TypeEnumerator.cs

```diff
@@ -97,8 +97,10 @@ internal Collection<UnitTestElement> GetTests(ICollection<string> warnings)
             // Test class is already valid. Verify methods.
             foreach (var method in this.type.GetRuntimeMethods())
             {
-                // Todo: Provide settings to allow users to pick up tests from other assemblies as well.
-                if (!method.DeclaringType.GetTypeInfo().Assembly.Equals(this.type.GetTypeInfo().Assembly))
+                var isMethodDeclaredInTestTypeAssembly = method.DeclaringType.GetTypeInfo().Assembly.Equals(this.type.GetTypeInfo().Assembly);
```

Comment:
>method.DeclaringType.GetTypeInfo().Assembly.Equals(this.type.GetTypeInfo().Assembly) [](start = 57, length = 84)

This seems to be causing issues in testability. How about putting this in a separate API in reflecthelper and mocking that out in your tests?